polaris-core/src/main/java/org/apache/polaris/core/PolarisConfigurationStore.java

```diff
@@ -68,6 +69,8 @@ public interface PolarisConfigurationStore {
 
     if (config.defaultValue instanceof Boolean) {
       return config.cast(Boolean.valueOf(String.valueOf(value)));
+    } else if (config.defaultValue instanceof List<?>) {
+      return config.cast(List.copyOf((List<?>) value));
```

Comment:
Maybe worth to have a config util set of methods to consolidate this usage (not in this PR though).

Reply:
Agreed, this is quite ugly to have in this class.